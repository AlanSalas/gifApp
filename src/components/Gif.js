import PropTypes from "prop-types";

const Gif = ({ title, url }) => {
  return (
    <div className="gif">
      <img src={url} alt={title} title={title} />
    </div>
  );
};

Gif.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Gif;
