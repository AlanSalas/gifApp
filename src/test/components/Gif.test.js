import { shallow } from "enzyme";
import Gif from "../../components/Gif";

describe("Tests in Gif component", () => {
  const title = "title";
  const url = "https://test.com";
  const wrapper = shallow(<Gif title={title} url={url} />);

  test("Should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Img should have src, title, alt", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
    expect(img.prop("title")).toBe(title);
  });
});
