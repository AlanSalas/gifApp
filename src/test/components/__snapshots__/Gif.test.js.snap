// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Gif component Should render properly 1`] = `
<div
  className="gif"
>
  <img
    alt="title"
    src="https://test.com"
    title="title"
  />
</div>
`;
